socialsharing.php

```diff
@@ -152,8 +152,9 @@ public function getContent()
 
 	public function hookDisplayHeader($params)
 	{
-		if (!isset($this->context->controller->php_self) || !in_array($this->context->controller->php_self, array('product', 'products-comparison')))
+		if (!isset($this->context->controller) || !method_exists($this->context->controller, 'getProduct')) {
```

Comment:
Hi,

This is changing behavior, does CompareController.php has a method getProduct ?

Regards